cli/src/commands/status.rs

```diff
@@ -91,21 +91,54 @@ pub(crate) fn cmd_status(
             writeln!(formatter)?;
         }
 
-        let wc_revset = RevsetExpression::commit(wc_commit.id().clone());
-        // Ancestors with conflicts, excluding the current working copy commit.
-        let ancestors_conflicts = workspace_command
-            .attach_revset_evaluator(
-                wc_revset
-                    .parents()
-                    .ancestors()
-                    .filtered(RevsetFilterPredicate::HasConflict)
-                    .minus(&revset_util::parse_immutable_expression(
-                        &workspace_command.revset_parse_context(),
-                    )?),
-            )?
-            .evaluate_to_commit_ids()?
-            .collect();
-        workspace_command.report_repo_conflicts(formatter, repo, ancestors_conflicts)?;
+        if wc_commit.has_conflict()? {
+            let wc_revset = RevsetExpression::commit(wc_commit.id().clone());
+
+            if let Some(first_descendant_no_conflict) = workspace_command
+                .attach_revset_evaluator(
+                    wc_revset
+                        .children()
+                        .descendants()
+                        .filtered(RevsetFilterPredicate::HasNoConflict),
+                )?
+                .evaluate_to_commits()?
+                .collect::<Vec<_>>()
+                .pop()
+            {
```

Comment:
Does it make sense to check wc children (which usually doesn't exist)? The user could `edit` a conflict commit, but in that case, I don't think they would be interested in whether descendants have a conflict or not.

Reply:
The rationale here is so that in the case  where a descendant has already resolved the conflict, printing out the same instructions to resolve the conflict again would be misleading.

This could happen if:

```
# wc has conflict 
$ jj new

# 1. resolve conflict, but don't squash yet 
# 2. now step back to previous commit for whatever reason 

$ jj prev --edit
$ jj status

# informing to resolve the conflict all over again here, is a bit confusing, since we know it has already been resolved. Probably better to state clearly that it's been resolved in a descendant. Though current message in the PR could be improved?
```

wdyt?

Reply:
I don't think it's a common mistake we need to catch. It's also unclear whether the wc child contains only conflict resolution. If a rebased commit happened to resolve conflict in parent for example, it may contain other unrelated changes, and the user will have to resolve the root conflict anyway.

Reply:
Makes sense. I've removed this. So all we're doing now is:

* If wc is conflict free:
  * if parent has a conflict, mention that wc resolves the conflict in the parent.
  * if parent has no conflict. do nothing.
* if wc has conflicts: print the same message we printed before, regardless of the state of the children.

This behaviour resolves the main complain in #4147 so I believe this is good enough.